aten/src/THCUNN/SpatialUpSamplingBicubic.cu

```diff
@@ -0,0 +1,243 @@
+#include "THCUNN.h"
+#include "THCTensor.hpp"
+#include "common.h"
+#include "linear_upsampling.h"
+#include "THCDeviceTensor.cuh"
+#include "THCDeviceTensorUtils.cuh"
+#include "THCDeviceUtils.cuh"
+#include "THCHalf.h"
+#include "THCHalfAutoNumerics.cuh"
+#include "THCAtomics.cuh"
+
+__constant__ int size = (1 << 10);
+
+template<typename Dtype>
+__device__ Dtype access_tensor(
+  const THCDeviceTensor<Dtype, 4> data,
+  int channel,
+  int batch,
+  int width,
+  int height,
+  int x,
+  int y
+) {
+  int access_x = max(min(x, width - 1), 0);
+  int access_y = max(min(y, height - 1), 0);
+  return data[batch][channel][access_y][access_x];
+}
+
+template<typename Dtype, typename Acctype>
+__device__ void inc_tensor(
+  const THCDeviceTensor<Dtype, 4> data,
+  int channel,
+  int batch,
+  int width,
+  int height,
+  int x,
+  int y,
+  Acctype value
+) {
+  int access_x = max(min(x, width - 1), 0);
+  int access_y = max(min(y, height - 1), 0);
+  atomicAdd(
+    data[batch][channel][access_y][access_x].data(),
+    ScalarConvert<Acctype, Dtype>::to(value)
+  );
+}
+
+template<typename Acctype>
+__device__ Acctype bicubic_convolution1(Acctype x, Acctype A) {
+  return ((A + 2) * x - (A + 3)) * x * x + 1;
+}
+
+template<typename Acctype>
+__device__ Acctype bicubic_convolution2(Acctype x, Acctype A) {
+  return ((A * x - 5 * A) * x + 8 * A) * x - 4 * A;
+}
+
+template<typename Acctype>
+__device__ void THNN_(get_coefficients)(
+  Acctype coeffs[4],
+  Acctype t,
+  int size
+) {
+  int offset = t * size;
+  Acctype A = -0.75;
```

Comment:
Please link these to math formulas you implemented. 
Actually I think A=-0.75 might be an argument that we want to expose on the Python side, since -0.5 and -0.75 are the most commonly used ones. What do you think? @fmassa @SsnL

Reply:
I agree with this opinion. Is there any specific reason for the choice of `-0.75` instead of [`-0.5` used in PIL](https://github.com/python-pillow/Pillow/blob/fdbd719da4c77c7e23e2e9e9b71d0d177f2d3369/src/libImaging/Resample.c#L45)? As `torchvision.transforms.functional.resize` uses PIL's `resize`, I may want to assume that the behavior of `interpolation` is the same as in `PIL` / `torchvision` as a PyTorch user.